### PR TITLE
feat: add barycentric weights to face annotations

### PR DESCRIPTION
**Pull request by WoodNeck**
## Details
- This adds barycentric weights to face annotations, allowing them to be put anywhere inside the face triangle.
- This adds handler for "webglcontextrestore" event, for restoring render infos.
- This enhances playground page of the demo website.

- https://woodneck.github.io/egjs-view3d

### CODE REVIEW

**Review by qpitlove — APPROVED** [review]
LGTM

File: src/const/browser.ts [hunk 1]
```diff
@@ -24,7 +24,9 @@ export const EVENTS = {
   MOUSE_LEAVE: "mouseleave",
   LOAD: "load",
   ERROR: "error",
-  CLICK: "click"
+  CLICK: "click",
+  CONTEXT_LOST: "webglcontextlost",
+  CONTEXT_RESTORE: "webglcontextrestored"
```
Comment (daybrush) [inline src/const/browser.ts:29]: It is okay to omit webgl. But I guess `RESTORED' would be nice?

**Review by daybrush — APPROVED** [review]
LGTM

File: src/const/browser.ts (hunk 1 above)
Comment (WoodNeck) [reply src/const/browser.ts:29]: Good point 👍

**Review by malangfox — APPROVED** [review]
LGTM

**Pull request merged**